cpp/CMakeLists.txt

```diff
@@ -320,6 +321,7 @@ if(BUILD_CUML_CPP_LIBRARY)
   target_link_libraries(${CUML_CPP_TARGET}
     PUBLIC
       rmm::rmm
+      cuml::Thrust
```

Comment:
Thrust targets need some some extra logic to be setup by consumers.

You can see how do this here:
https://github.com/rapidsai/cugraph/pull/1585/files#diff-1bba462ab050e89360fd88110a689e85ee037749cea091a1848ab574381d3795R352

Reply:
If that link doesn't work:

```cmake
set(code_string[=[thrust_create_target(cugraph::Thrust FROM_OPTIONS)]=]) 
rapids_export(INSTALL cugraph    
                       EXPORT_SET cugraph-exports
                       GLOBAL_TARGETS cugraph
                       NAMESPACE cugraph::
                       DOCUMENTATION doc_string
                       FINAL_CODE_BLOCK code_string    )
```

Reply:
After this change, I'm this error configuring with an RMM source-build dir:
```
CMake Error at /home/ptaylor/dev/rapids/rmm/cmake/install/FindThrust.cmake:59 (add_library):
  add_library cannot create target "thrust_internal" because another target
  with the same name already exists.  The existing target is an interface
  library created in source directory "/home/ptaylor/dev/rapids/cuml/cpp".
  See documentation for policy CMP0002 for more details.
```

Reply:
Seems like the above error is fixed by https://github.com/rapidsai/rmm/pull/784